### PR TITLE
Extract JavaFX and ensure JDK 16 compatibility

### LINKED ISSUE #31: Extract classes that depend on JavaFX
We only have 2 components that depend on JavaFX. Pulling in JavaFX only for those two seems like a heavy burden. I think we're better off if we extract those two components into a separate dependency.

### LINKED ISSUE #32: Make xapi fully runtime compatible with JDK 17
XAPI works with JDK 16 for most use cases. There are only a few classes that were moved in the last Java releases. If you don't call classes that depend on the moved classes you're fine. Let's fix that so you can call all classes again. 

The moved classes are: 
- javax.annotation.Generated
- com.sun.java.swing.plaf.nimbus
- all the JavaFX stuff (see https://github.com/xdev-software/xapi/issues/31)

### PR DESCRIPTION
**Pull request by rfichtner**
Fixes #32 
Fixes #31 

Also check out https://github.com/xdev-software/xapi-fx

### CODE REVIEW

**Review by AB-xdev — CHANGES_REQUESTED** [review]
Todo, due to breaking changes:
- [x] Update the version with a major jump -> 6.0.0-SNAPSHOT
- [x] Migration guide?

File: src/test/java/xdev/ui/tree/XdevTreeNodeTest.java [hunk 1]
```diff
@@ -51,6 +59,23 @@ private XdevTreeNode createTestNodes(){
...
+		XdevTreeNode node1 = (XdevTreeNode) nodes.get(0);
+		XdevTreeNode node2 = (XdevTreeNode) nodes.get(1);
+		
+		Assert.assertEquals("root", node1.getCaption());
+		Assert.assertEquals("level1", node2.getCaption());
 	}
+	
+	
+	@Test
+	public void searchTreeNodes(){
```
Comment (AB-xdev) [inline src/test/java/xdev/ui/tree/XdevTreeNodeTest.java]: This doesn't look like our "normal" formatter

File: src/main/java/xdev/ui/laf/NimbusLookAndFeel.java [hunk 2]
```diff
@@ -33,8 +33,14 @@ public void setLookAndFeel() throws LookAndFeelException
 	{
 		try
 		{
-			UIManager.setLookAndFeel(com.sun.java.swing.plaf.nimbus.NimbusLookAndFeel.class
-					.getName());
+			if (System.getProperty("java.version").startsWith("1.")) {
+				// Java 8 and below
+				UIManager.setLookAndFeel("com.sun.java.swing.plaf.nimbus.NimbusLookAndFeel");
+			}else {
```
Comment (AB-xdev) [inline src/main/java/xdev/ui/laf/NimbusLookAndFeel.java]: Also here the format seems to be wrong.
Please recheck all modified files.

**Review by AB-xdev — CHANGES_REQUESTED** [review]

File: src/test/java/xdev/ui/tree/XdevTreeNodeTest.java (hunk 1 above)
Comment (rfichtner) [reply src/test/java/xdev/ui/tree/XdevTreeNodeTest.java]: Fixed :)

File: src/main/java/xdev/ui/laf/NimbusLookAndFeel.java (hunk 2 above)
Comment (rfichtner) [reply src/main/java/xdev/ui/laf/NimbusLookAndFeel.java]: Checked and fixed all files 👍

File: README.md [hunk 3]
```diff
@@ -24,6 +24,20 @@ XDEV(-IDE) is a visual Java development environment for fast and easy applicatio
...
 The XDEV-IDE was license-free up to version 4 inclusive and is available for Windows, Linux and macOS. From version 5, the previously proprietary licensed additional modules are included in the IDE and the license of the entire product has been converted to a paid subscription model. The XDEV Application Framework, which represents the core of the RAD concept of XDEV and is part of every XDEV application, was released as open-source in 2008.
 
+## Migration Guide to version 6.0.0
+*This guide only applies if you use XAPI outside of the XDEV IDE.*
+### Dependencies
+If your project uses the components `XdevBrowser` or `XdevJFXPanel` you need to add a dependency to your project:
+
+    <dependency>
+        <groupId>com.xdev-software</groupId>
+        <artifactId>xapi-fx_Java8</artifactId>
```
Comment (AB-xdev) [inline README.md:34]: I think ``-java8`` should be at the end of the version

**Pull request merged**